earthly/python/Earthfile

```diff
@@ -22,23 +28,40 @@ python-base:
 
     # Stop python code running in the container saving pointless .pyc files.
     ENV PYTHONDONTWRITEBYTECODE=1
-
+    
     # Install Poetry to manage python.
     RUN curl -sSL https://install.python-poetry.org | python3 -
     # Prevent warning when running on machines with more than 6 cores.
     #   "Connection pool is full, discarding connection: pypi.org. Connection pool size: 10"
     RUN poetry config installer.max-workers 10
+    
+    # Install flake8 globally using Poetry
+    RUN pip3 install flake8
 
-# Common python builder setup.
-# Parameters:
-#   opts      : Options passed to the `poetry install` command.
 BUILDER:
     FUNCTION
-    ARG opts
+    ARG opts=""
+
+    FROM +python-base
 
     # Copy our dependencies.
     COPY pyproject.toml poetry.lock .
 
     # Install it all with poetry
     RUN poetry install $opts
 
+PYTHON_LINT:
+    FUNCTION
+    # Optionally, specify the directory containing Python code to lint.
+    ARG src_dir=./src
+
+    FROM +python-base
+
+    # Copy the source code directory to the container.
+    COPY $src_dir /app
+
+    WORKDIR /app
+
+    # Run flake8 to lint the Python code.
+    # Adjust flake8 options as needed.
+    RUN flake8 .
```

Comment:
ruff
```suggestion
    RUN ruff check .
```

Reply:
Changed to ruff commit c90df2f16b81271f0563faeccfa1d25161ca46af